### PR TITLE
create bestbuy demo

### PR DESCRIPTION
**Pull request by tuanddd**

### CODE REVIEW

File: demos/bestbuy/src/app.tsx [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        return `${min} - ${max}`;
+      }
+      case v.startsWith('>'):
+        return `Over ${formatNumber(Number(v.substring(2)), 'USD', true)}`;
+      case v.startsWith('<'):
+        return `Under ${formatNumber(Number(v.substring(2)), 'USD', true)}`;
+      default:
+        return null;
+    }
+  }, []);
```
Comment (sampotts) [inline demos/bestbuy/src/app.tsx:44]: I think we should build this into the React SDK as I did for the `<RangeInput />` where you can specify `format` being `'numeric'` or `'price'`. What do you think?

File: demos/bestbuy/src/app.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+@tailwind components;
+/* purgecss end ignore */
+
+@tailwind utilities;
+
+/* Nicer check for checkbox */
+.form-checkbox:checked {
+  background-image: url("data:image/svg+xml,%3Csvg fill='white' viewBox='0 0 16 16' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='m6.9919558 12.0005941c-.26547704 0-.52006014-.1054507-.7077524-.293l-2.00319264-2.00000003c-.37937552-.39237889-.37394992-1.01608478.01219439-1.40181858.3861443-.38573379 1.01051394-.39115363 1.40331041-.01218142l1.20234127 1.2 3.30864357-5.107c.3394637-.43630494.9687248-.51510281 1.4054941-.17599998.4367692.33910283.515651.96769503.1761873 1.40399998l-4.00638527 6.00000003c-.1761023.2273124-.44160568.3679379-.72877475.386z'/%3E%3C/svg%3E");
+}
```
Comment (sampotts) [inline demos/bestbuy/src/app.css:12]: Shouldn't we just use `import '@sajari-ui/core';` here? Would need to add `postcss-import`.

File: demos/bestbuy/src/app.css [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/* purgecss start ignore */
```
Comment (sampotts) [inline demos/bestbuy/src/app.css:1]: I don't think these comments are required any more. I could be wrong. Worth a check.

File: demos/bestbuy/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import App from './app';
+import { SearchProvider } from '@sajari/react-search-ui';
+import { Pipeline, FilterBuilder } from '@sajari/react-hooks';
+import { h } from 'preact';
+
+const pipeline = new Pipeline(
+  {
+    account: '1594153711901724220',
+    collection: 'bestbuy',
+    endpoint: 'https://jsonapi-us-valkyrie.sajari.net',
```
Comment (sampotts) [inline demos/bestbuy/src/index.tsx:12]: I think we can omit this

File: demos/bestbuy/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  },
+  count: false,
+});
+
+export default () => (
+  <SearchProvider
+    search={{
+      pipeline,
+      // @ts-ignore
+      fields: { category: (data) => data.level4 || data.level3 || data.level2 || data.level1, title: 'name' },
```
Comment (sampotts) [inline demos/bestbuy/src/index.tsx:58]: Does TS complain about data?

File: demos/bestbuy/src/number.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    return new Intl.NumberFormat(navigator.language).format(input);
+  }
+
+  const lang = neutral ? navigator.language.split('-')[0] : navigator.language;
+
+  return new Intl.NumberFormat(lang, {
+    style: 'currency',
+    currency: currencyCode,
+  }).format(input);
+}
```
Comment (sampotts) [inline demos/bestbuy/src/number.ts:22]: Perhaps we can use `formatNumber` from `@sajari/react-utils`?

File: demos/bestbuy/tests/__mocks__/browserMocks.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+		getItem: (key) => store[key] || null,
+		setItem: (key, value) => store[key] = value.toString(),
+		clear: () => store = {}
+	};
+
+})();
+
+Object.defineProperty(window, 'localStorage', {
+	value: localStorageMock
+}); */
```
Comment (sampotts) [inline demos/bestbuy/tests/__mocks__/browserMocks.js:21]: Do we need this? Also, it's using 4 space tabs.

File: demos/bestbuy/tests/header.test.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+// See: https://github.com/preactjs/enzyme-adapter-preact-pure
+import { shallow } from 'enzyme';
+
+describe('Initial Test of the Header', () => {
+	test('Header renders 3 nav items', () => {
+		const context = shallow(<Header />);
+		expect(context.find('h1').text()).toBe('Preact App');
+		expect(context.find('Link').length).toBe(3);
+	});
+});
```
Comment (sampotts) [inline demos/bestbuy/tests/header.test.js:12]: Do we need this? Also, it's using 4 space tabs.

File: demos/bestbuy/tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,59 @@
+{
+  "compilerOptions": {
+    /* Basic Options */
+    "target": "ES5" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', or 'ESNEXT'. */,
+    "module": "ESNext" /* Specify module code generation: 'none', commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */,
+    "lib": ["dom", "esnext"] /* Specify library files to be included in the compilation:  */,
+    "allowJs": true /* Allow javascript files to be compiled. */,
+    // "checkJs": true,                       /* Report errors in .js files. */
+    "jsx": "preserve" /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */,
```
Comment (sampotts) [inline demos/bestbuy/tsconfig.json:9]: I think we can use `"react-jsx"` now with 4.1+

```
"jsx": "react-jsx",
"jsxImportSource": "preact",
```

and omit the `"jsxFactory"`.

File: demos/bestbuy/src/app.tsx [hunk 10]
```diff
@@ -0,0 +1,243 @@
+/** @jsx h */
```
Comment (sampotts) [inline demos/bestbuy/src/app.tsx:1]: I think we can ditch this with some changes to the tsconfig.json, adding `@babel/plugin-transform-react-jsx` and then adding this to `preact.config.js`: 

```js
const { rule } = helpers.getLoadersByName(config, 'babel-loader')[0];
const babelConfig = rule.options;

babelConfig.plugins.push([
  require.resolve('@babel/plugin-transform-react-jsx'),
  {
    pragma: 'h',
  },
]);
```

File: demos/bestbuy/postcss.config.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable global-require, import/no-extraneous-dependencies */
+const { tailwindConfig } = require('@sajari-ui/core');
+
+// Setup purging of CSS
+tailwindConfig.purge.content.push('./src/**/*.{js,html}');
+
+// Add full padding option for images
+tailwindConfig.theme.extend.padding = {
+  full: '100%',
+};
```
Comment (sampotts) [inline demos/bestbuy/postcss.config.js:10]: Don't think we'd need this.

**Review by sampotts — COMMENTED** [review]
Is there any way I can view this built other than pulling it down and building it?

File: demos/bestbuy/src/app.tsx [hunk 12]
```diff
@@ -0,0 +1,243 @@
...
+    }
+
+    if (results && results.length === 0) {
+      return (
+        <Flex flexDirection="flex-col" space="space-y-2" alignItems="items-center">
+          <Heading size="md">No results</Heading>
+          <Text>Sorry, we couldn&apos;t find any matches.</Text>
+        </Flex>
+      );
+    }
```
[the remark this answers is not in the document]
Comment (tuanddd) [reply demos/bestbuy/src/app.tsx:63]: Yeah this is quite old now, will update it.

File: demos/bestbuy/src/app.tsx (hunk 1 above)
Comment (tuanddd) [reply demos/bestbuy/src/app.tsx:44]: Good point, will do that.

File: demos/bestbuy/src/app.tsx (hunk 10 above)
Comment (tuanddd) [reply demos/bestbuy/src/app.tsx]: Setting the above `tsconfig.json` and this will result in duplicated preset.

Comment (sampotts) [reply demos/bestbuy/src/app.tsx]: OK well whichever way we can ditch the need for the pragma 👍

**Pull request closed without merging**